key_action_map.cpp

```diff
@@ -163,6 +168,82 @@ sf::Keyboard::Key get_stun_key(const key_action_map& m)
   return m.to_key(action_type::shoot_stun_rocket);
 }
 
+bool exists_file (const std::string& name) {
+    std::ifstream f(name.c_str());
+    return f.good();
+}
+
+key_action_map load_kam(const std::string& filename)
+{
+  std::ifstream f(filename);
+  key_action_map m;
+  f >> m;
+  return m;
+}
+
+void save_to_file(const key_action_map& kam, const std::string& filename)
+{
+  std::ofstream file(filename);
+  file << kam;
+}
+
+sf::Keyboard::Key to_sfml_key(const std::string& s)
+{
+  if (s == "A") return sf::Keyboard::Key::A;
+
+
+  assert(s == " "); // Or the key has not been encoded yet
+  return sf::Keyboard::Key::Space;
+
+}
+
+std::string to_str(const key_action_map& kam) noexcept
+{
+  std::stringstream s;
+  for (const auto& p: kam.get_raw_map()) {
+    s << p.first << " " << p.second << " ";
```

Comment:
Seems like `std::maps` sorts its elements in lexicographic order (alphabetic order) according to the first key. This loop will not generate a string based on the insertion order.

Reply:
`0 turn_left 3 turn_right 4 shoot_stun_rocket 16 shoot 18 accelerate_backward 22 accelerate_forward`
This is the actual order in a saved file

Reply:
I'll take a look :-)